plugin/src/test/java/hudson/plugins/swarm/PipelineJobTest.java

```diff
@@ -0,0 +1,132 @@
+package hudson.plugins.swarm;
+
+import static org.junit.Assert.assertNotNull;
+
+import com.google.common.collect.Iterables;
+import hudson.model.Node;
+import hudson.plugins.swarm.test.ProcessDestroyer;
+import hudson.plugins.swarm.test.TestUtils;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.Assume;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+import org.jvnet.hudson.test.TemporaryDirectoryAllocator;
+
+public class PipelineJobTest {
+
+    @Rule public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    private final ProcessDestroyer processDestroyer = new ProcessDestroyer();
+    private final TemporaryDirectoryAllocator tmpDir = new TemporaryDirectoryAllocator();
+
+    /** Executes a shell script build on a Swarm Client agent. */
+    @Test
+    public void buildShellScript() throws Exception {
+        story.addStep(
+                new Statement() {
+                    @Override
+                    public void evaluate() throws Throwable {
+                        Node node =
+                                TestUtils.createSwarmClient(
+                                        story.j, processDestroyer, tmpDir.allocate());
+
+                        WorkflowJob project = story.j.createProject(WorkflowJob.class);
+                        project.setConcurrentBuild(false);
+                        project.setDefinition(new CpsFlowDefinition(getFlow(node, 0), true));
+
+                        WorkflowRun build = project.scheduleBuild2(0).waitForStart();
+                        story.j.waitForCompletion(build);
+                        story.j.assertLogContains("ON_SWARM_CLIENT=true", build);
+                        story.j.assertBuildStatusSuccess(build);
+                        tearDown();
+                    }
+                });
+    }
+
+    /**
+     * Starts a Jenkins job on a Swarm Client agent, restarts Jenkins while the job is running, and
+     * verifies that the job continues running on the same agent after Jenkins has been restarted.
+     */
+    @Test
+    public void buildShellScriptAfterRestart() throws Exception {
+        Assume.assumeNotNull(
+                System.getProperty("port"), "This test requires a fixed port to be available.");
+
+        final AtomicReference<String> projectName = new AtomicReference<>();
```

Comment:
Unnecessary; you can just pass a `name` argument to `createProject`.

Reply:
Done.